pkg/memlimit/limit_linux.go

```diff
@@ -0,0 +1,101 @@
+package memlimit
+
+// This file is heavily inspired by Clair:
+// https://github.com/quay/clair/blob/v4.7.2/initialize/auto/memory_linux.go.
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"runtime/debug"
+	"strconv"
+)
+
+// setMemoryLimit sets the runtime's memory limit based on information gleaned from the
+// current process's cgroup. See [debug.SetMemoryLimit] for details on the effects
+// of setting the limit.
+//
+// The process' "memory.max" limit (for cgroups v2) or
+// "memory.limit_in_bytes" (for cgroups v1) are the values consulted.
+func setMemoryLimit() int64 {
+	root := os.DirFS("/")
+	lim, err := memLookup(root)
+	switch {
+	case err != nil:
+		return 0
+	case lim == doNothing || lim == setMax:
+		return 0
+	}
+	// Following the GC guide and taking a haircut: https://tip.golang.org/doc/gc-guide#Suggested_uses
+	tgt := lim - (lim / 20)
+	debug.SetMemoryLimit(tgt)
+	return tgt
+}
+
+const (
+	doNothing = -1
+	setMax    = -2
+)
+
+func memLookup(r fs.FS) (int64, error) {
```

Comment:
this function looks complicated. Why not using a lib to parse cgroup data? E.g.  https://github.com/containerd/cgroups

Reply:
Going to just remove this in favor of the env var approach